dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -119,6 +119,30 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
     }
   }
 
+  def createNewRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      start: Int,
+      stop: Int,
+      step: Int = 1): Future[EventDb] = {
```

Comment:
I would just add requires upfront saying

```scala
require(start >= 0)
require(stop >= start)
require(step > 0)
```

With appropriate error messages

Reply:
I don't see the reason in disallowing descending values or negative values

Reply:
Ended up going with this.